datalogger/logger.go

```diff
@@ -62,10 +74,10 @@ func (logger *CSVDataLogger[T]) Export() bool {
 		return false
 	}
 
-	t := new(T)
-	visibleFields := reflect.VisibleFields(reflect.TypeOf(t).Elem())
+	visibleFields := reflect.VisibleFields(reflect.TypeOf((*T)(nil)).Elem())
```

Comment:
Woah. I've never seen this before. Where did you find it?

Reply:
https://pkg.go.dev/reflect#example-TypeOf

Reply:
Huh! Looks like they are just casting `nil` to a pointer to the type that they want. I think that works because they are expecting an Interface type which are more "robust" against `nil`s.